aws/resource_aws_batch_job_queue_test.go

```diff
@@ -104,29 +106,88 @@ func TestAccAWSBatchJobQueue_disappears(t *testing.T) {
 	})
 }
 
-func TestAccAWSBatchJobQueue_update(t *testing.T) {
-	var jq batch.JobQueueDetail
-	ri := acctest.RandInt()
-	config := fmt.Sprintf(testAccBatchJobQueueBasic, ri)
-	updateConfig := fmt.Sprintf(testAccBatchJobQueueUpdate, ri)
-	resourceName := "aws_batch_job_queue.test_queue"
+// Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/8083
+func TestAccAWSBatchJobQueue_ComputeEnvironments_ExternalOrderUpdate(t *testing.T) {
+	var jobQueue1 batch.JobQueueDetail
```

Comment:
Personally I find appending the number from the start helps reduce code churn should the test be extended to need multiple later on, but we'll want to setup something to enforce the naming of these if its a concern.

Reply:
We could also do something like always require it to be a slice, e.g. `var jobQueues []*batch.JobQueueDetail`, then adjust `...Exists` functions to append and ones reading the information like `...Recreated` functions to require a two index parameters, e.g. `...Recreated(jobQueues, 0, 1)`

Reply:
Even better, honestly, would be not handling the API object at all in most cases. Its typically added now to avoid the churn in case its needed later on, but we could switch up the testing a little to enable generic versions of the various functions like `...Recreated`/`...NotRecreated` that use attributes in the state across `TestStep`. It'd be best if the SDK provided that functionality though.

Reply:
Sorry for all the comments here. 😄 Here's a reference implementation to my previous comment: https://github.com/hashicorp/terraform-plugin-sdk/pull/329